### PR TITLE
Refactor Remaining string modules

### LINKED ISSUE #170: Fix branding
I think we have the branding a little backwards... We want the key to be a symbol, and the value can be anything that also has a symbol inside of it.

cf io-ts `Brand` https://github.com/gcanti/io-ts/blob/master/src/index.ts#L1132-L1162 which I think we should use.

This also lets us put metadata about the brand on it, if we so choose

### LINKED ISSUE #174: `HslColor`

### LINKED ISSUE #175: `RGB`

### LINKED ISSUE #176: `UUID`

### LINKED ISSUE #177: `LatLong`

### PR DESCRIPTION
**Pull request by skeate**
- Adds trailing commas rule in Prettier. I kept yanking lines around and then having things break because they lacked trailing commas.
- `HslColor` (closes #174)
- `RGB` (closes #175)
- `UUID` (closes #176)
- `LatLong` (closes #177)
- Updated branding with a relevant usage in the `UUID` module (the version is now part of the brand) (closes #170)
- Removed schemable generation code that relied on `src/string` since that folder no longer exists and we have no plans to ever remake it (new modules should be `WithX` extensions)
- A functional `integerRange` for pattern builder! woo
   - `regexFromPattern(PB.integerRange(12, 31))` => (something like)  `/^(1[2-9]|2\d|3[0-1])$/`

apologies for the length, but the commits should be _mostly_ self-contained.

### CODE REVIEW

File: src/schemata/string/UUID.ts [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+ */
+export type UUID<Version extends UUIDVersion> = Branded<string, UUIDBrand<Version>>
+
+/**
+ * @since 1.0.0
+ * @category Model
+ */
+export type UUIDS<Version extends UUIDVersion> = SchemaExt<string, UUID<Version>>
+
+export const UUID = <Version extends UUIDVersion>(version: Version): UUIDS<Version> =>
```
Comment (jacob-alford) [inline src/schemata/string/UUID.ts:139]: You might need a jsdoc here for docs-ts, though jsdoc will be merged in vscode with all exports with identical names (not sure how it works)

File: tests/PatternBuilder.test.ts [hunk 2]
```diff
@@ -60,4 +60,121 @@ describe('PatternBuilder', () => {
...
+          } else {
+            expect([i.toString(), actual.test(i.toString())]).toEqual([
+              i.toString(),
+              false,
+            ])
+          }
+        }
+      })
+
+      test('arbitrary ranges', () => {
```
Comment (jacob-alford) [inline tests/PatternBuilder.test.ts:155]: Great tests!

**Review by jacob-alford — APPROVED** [review]
Looks great!  Just needs a quick jsdoc fix and it should be good to merge.  

Also, this might close #92, but not completely sure.

File: src/schemata/string/UUID.ts (hunk 1 above)
Comment (skeate) [reply src/schemata/string/UUID.ts:139]: oops

**Comment by skeate** [comment]
it does not close #92

**Review by jacob-alford — APPROVED** [review]
LGTM!

**Pull request merged**